src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/Razor/RazorDynamicFileInfoProvider.cs

```diff
@@ -136,13 +86,17 @@ private ValueTask UpdateAsync(ImmutableSegmentedList<string> paths, Cancellation
         return ValueTask.CompletedTask;
     }
 
-    private sealed class EmptyStringTextLoader() : TextLoader
+    private async Task<IRazorLSPDynamicFileInfoProvider> GetFileProviderAsync(CancellationToken cancellationToken)
     {
-        public static readonly TextLoader Instance = new EmptyStringTextLoader();
+        var dynamicFileInfoProvider = await RazorLSPServiceProvider.GetRequiredServiceAsync<IRazorLSPDynamicFileInfoProvider>(cancellationToken);
 
-        public override Task<TextAndVersion> LoadTextAndVersionAsync(LoadTextOptions options, CancellationToken cancellationToken)
-        {
-            return Task.FromResult(TextAndVersion.Create(SourceText.From(""), VersionStamp.Default));
-        }
+        await dynamicFileInfoProvider.EnsureInitializedAsync(
+            static ct =>
+            {
+                return RazorLSPServiceProvider.GetRequiredServiceAsync<IRazorClientLanguageServerManager>(ct);
```

Comment:
Can you add a comment explaining this indirection? Seems odd to initialize a razor service with a delegate to get a different razor service from a razor service provider. Can the initialize method not take the `IRazorClientLanguageServerManager` directly? Or even better, the `IRazorServiceProvider` in case its dependencies change?